cartesi-rollups/development/assets-handling/overview.md

```diff
@@ -0,0 +1,74 @@
+---
+id: overview
+title: Overview
+tags: [assets, portals, vouchers]
+resources:
+  - url: https://www.udemy.com/course/the-cartesi-dapp-developer-masterclass
+    title: The Cartesi dApp Developer Free Course
+  - url: https://github.com/jjhbk/cartesi-wallet
+    title: "Cartesi-Wallet: TypeScript-based Wallet Implementation for Cartesi dApps"
+  - url: https://github.com/jjhbk/cartesi-router
+    title: "Cartesi-Router: TypeScript-based Router Implementation for Cartesi dApps"
+  - url: https://github.com/Mugen-Builders/frontend-cartesi-wallet-x
+    title: "Frontend-Cartesi-Wallet: A React frontend to showcase wallet functionality of Cartesi dApps"
+---
+
+Assets exist on the base layer, where they have actual meaning and value.
+
+As with any execution layer solution, a Cartesi dApp that wants to manipulate assets needs a secure way of "teleporting" the assets from the base layer to the execution layer and then a way to "teleport" them back to the base layer.
+
+Currently, Cartesi Rollups support the following types of assets:
+
+- [Ether (ETH)](../../core-concepts/rollup-http-api/json-rpc/portals/EtherPortal.md)
+- [ERC-20](../../core-concepts/rollup-http-api/json-rpc/portals/ERC20Portal.md)
+- [ERC-721](../../core-concepts/rollup-http-api/json-rpc/portals/ERC721Portal.md)
+- [ERC-1155 Single](../../core-concepts/rollup-http-api/json-rpc/portals/ERC1155SinglePortal.md)
+- [ERC-1155 Batch](../../core-concepts/rollup-http-api/json-rpc/portals/ERC1155BatchPortal.md)
+
+![img](../../../static/img/v1.3/assets.jpg)
+
+## Deposits
+
+Portals enable the safe transfer of assets from the base layer to the execution layer. Users authorize portals to deduct assets from their accounts and initiate transfers to dApps.
+
+Once deposited, those L1 assets gain a representation in the execution layer and are owned by whomever the depositor assigned them to.
+
+When an asset is deposited, the corresponding Portal contract sends an input via the InputBox contract describing the type of asset, amount, and some data the depositor might want the dApp to read. The off-chain machine will then interpret and validate the input payload.
+
+Deposit input payloads are always specified as packed ABI-encoded parameters, as detailed below.
+
+### ABI encoding for deposits
```

Comment:
I would add one example of decoding here, using any higher-level language or pseudo-code. It could be ERC20 transfer. Should we also add links to the sourcecode? [ERC20 example](https://github.com/cartesi/rollups-contracts/blob/77ff2c08928b871348b6e2bf3b6f5d7f3c95c9dc/onchain/rollups/contracts/portals/ERC20Portal.sol#L30)

Reply:
I'm going to put the Python code there for now.  Lets circle back and think through this later

Reply:
I have added a large python block of code. Nevertheless, we need to circle back on this